src/main/java/us/abstracta/wiresham/ReloadService.java

```diff
@@ -0,0 +1,103 @@
+package us.abstracta.wiresham;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReloadService implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ReloadService.class);
+  private final VirtualTcpService service;
+  private final ExecutorService reloadExecutorService =
+      Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
+          .setNameFormat("Auto-Reload-Service-%d").build());
+  private final File configFile;
+  private boolean registered;
+
+  public ReloadService(VirtualTcpService service, File configFile) {
+    this.service = service;
+    this.configFile = configFile;
+  }
+
+  @Override
+  public void run() {
+    Path watchServicePath = getAbsolutFlowPath();
+    WatchService watchService;
+    try {
+      watchService = FileSystems.getDefault().newWatchService();
+      watchServicePath.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);
+      registered = true;
+    } catch (IOException e) {
+      LOG.error("Error while initializing Reload Service", e);
+      return;
+    }
+    while (true) {
+      try {
+        WatchKey take = watchService.take();
+
+        if (!take.isValid()) {
```

Comment:
capaz conviene hacer un metodo para procesar el take? y meter el while dentro del catch, asi te ahorras el return y probablemente quede mas modularizado. Inclusive podrias meter en el run el if y for y hacer un metodo solo para el pollEvent (que capaz hasta tiene mas sentido). Asi como esta con un catch con un return y una variable colgada antes del while, queda medio raro, capaz creando un metodo para la logica interior y modularizarlo ayuda?